app/controllers/agent/eligibility/SoleTraderController.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.agent.eligibility
+
+import java.time.LocalDate
+
+import auth.agent.StatelessController
+import config.AppConfig
+import forms.agent.SoleTraderForm._
+import javax.inject.{Inject, Singleton}
+import models._
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.Security.Authenticated
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Request}
+import services.AuthService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.play.language.LanguageUtils
+import utilities.ImplicitDateFormatter
+import views.html.agent.eligibility.are_you_a_sole_trader
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SoleTraderController @Inject()(val authService: AuthService)
+                                    (implicit appConfig: AppConfig,
+                                     mcc: MessagesControllerComponents,
+                                     override val languageUtils: LanguageUtils,
+                                     val ec: ExecutionContext) extends StatelessController with I18nSupport with ImplicitDateFormatter {
+
+  private def startDateLimit: LocalDate = LocalDate.now.minusYears(2)
+
+  private def form(startDate: LocalDate)(implicit request: Request[_]): Form[YesNo] = soleTraderForm(startDate.toLongDate)
+
+  def backUrl: String = routes.OtherSourcesOfIncomeController.show().url
+
+  def show: Action[AnyContent] = Authenticated { implicit request =>
+    implicit user =>
+      Ok(are_you_a_sole_trader(form(startDateLimit), routes.SoleTraderController.submit(), startDateLimit.toLongDate, backUrl))
+  }
+
+  def submit(): Action[AnyContent] = Action.async {
```

Comment:
```suggestion
  def submit(): Action[AnyContent] = Authenticated {
```
This should be Authenticated, making this change, you will also be able to remove Future.successful from each result